pkg/registry/common/setregistrationtime/nse_server.go

```diff
@@ -19,15 +21,19 @@ package setregistrationtime
 import (
 	"context"
 
+	"github.com/edwarnicke/genericsync"
 	"github.com/golang/protobuf/ptypes/empty"
 	"github.com/networkservicemesh/api/pkg/api/registry"
+	"google.golang.org/protobuf/proto"
 	"google.golang.org/protobuf/types/known/timestamppb"
 
 	"github.com/networkservicemesh/sdk/pkg/registry/core/next"
 	"github.com/networkservicemesh/sdk/pkg/tools/clock"
 )
 
-type setregtimeNSEServer struct{}
+type setregtimeNSEServer struct {
+	genericsync.Map[string, *timestamppb.Timestamp]
```

Comment:
Can we use metadata instead?

Reply:
@denis-tingaikin 
I thought about it, but unfortunately we don't have `metadata` for registry

Reply:
It might be a good time to add it ;)

Reply:
Fixed :)